packages/@aws-cdk/aws-lambda/README.md

```diff
@@ -277,6 +277,42 @@ correct log retention period (never expire, by default).
 *Further note* that, if the log group already exists and the `logRetention` is not set, the custom resource will reset
 the log retention to never expire even if it was configured with a different value.
 
+### FileSystem Access
+
+You can configure a function to mount an Amazon Elastic File System (Amazon EFS) to a
+directory in your runtime environment with the `filesystems` property. To access Amaozn EFS
+from lambda function, the Amazon EFS access point will be required. 
+
+Amazon EFS _access points_ are application-specific entry points into an EFS file system that
+make it easier to manage application access to shared datasets. Access points can enforce a user
+identity, including the user's POSIX groups, for all file system requests that are made through
+the access point. Access points can also enforce a different root directory for the file system
+so that clients can only access data in the specified directory or its subdirectories. See
+[Working with Amazon EFS Access Points](https://docs.aws.amazon.com/efs/latest/ug/efs-access-points.html) for more details.
+
+The following sample enables the lambda function to mount the Amazon EFS filesystem via the `accessPoint` to `/mnt/msg` in the runtime environment.
+
+```ts
+const fileSystem = new efs.FileSystem(stack, 'Efs', { vpc });
+
+const accessPoint = fileSystem.addAccessPoint('AccessPoint');
+
+const fn = new lambda.Function(stack, 'MyLambda', {
+  code,
+  handler,
+  runtime,
+  vpc,
+  securityGroups: fileSystem.connections.securityGroups,
+  filesystems: [{
```

Comment:
I don't see a reason to separate the target and the mount path to form the file system configuration. It feels unnecessarily convoluted. Can you talk to why my original proposal did not work?

```ts
new lambda.Function(stack, 'MyLambda', {
  fileSystems: [ 
    lambda.FileSystem.fromEfsAccessPoint(accessPoint, '/mnt/fs1'[, options]),

    // should we also provide a sugar that just accepts an EFS file system directly and defines the AP for you?
    lambda.FileSystem.fromEfsFileSystem(fileSystem, '/mnt/fs2'[, options])
  ]
});
```

This is a very common pattern in the CDK where we model a data structure as "a union like class". 

```ts
class FileSystem {
  public static fromEfsAccessPoint(ap: efs.AccessPoint, nountPath: string, options: EfsAccessPointOptionns = {}): FileSystem {
    return new FileSystem({
      LocalMountPath: mountPath, 
      Arn: ap.accessPointArn
    });
  }

  protected constructor(public readonly config: any) { }
}
```

Something like this...

Reply:
OK. Let me revise it. Thanks for the chime in.

Reply:
Hi @eladb , I've updated it accordingly. Please check it out again.

And I exposed the `accessPoint.filesystem` from efs construct lib now. Still trying to figure out how to `addDependency` on it in lambda.Function.

However, as `fromEfsAccessPoint` only returns 

```
{
  LocalMountPath: ...
  Arn: ...
}
``` 

I was wondering if we should return something like `dependency` as an array like this
```
{
  localMountPath: ...
  arn: ...
  dependency: [...]
}
```

so the lambda.Function can simply `addDepencency()` on them. And in this case `accessPoint.filesystem` only.  Thoughts?